app/models/miq_server/server_smart_proxy.rb

```diff
@@ -69,26 +69,12 @@ def queue_call(ost)
     case ost.method_name
     when "scan_metadata", "sync_metadata"
       worker_setting = MiqSmartProxyWorker.worker_settings
-      #
-      # TODO: until we get location/offset read capability for OpenStack
-      # image data, OpenStack fleecing is prone to timeout (based on image size).
-      # We try to adjust the timeout here - maybe this should be calculated based
-      # on the size of the image, but that information isn't directly available.
-      #
-      timeout_adj = 1
+
+      timeout_adjustment_multiplier = 1
       if ost.method_name == "scan_metadata"
         klass = ost.target_type.constantize
         target = klass.find(ost.target_id)
-        if target.kind_of?(ManageIQ::Providers::Openstack::CloudManager::Vm) ||
-           target.kind_of?(ManageIQ::Providers::Openstack::CloudManager::Template)
-          timeout_adj = 4
-        elsif target.kind_of?(ManageIQ::Providers::Azure::CloudManager::Vm) ||
-              target.kind_of?(ManageIQ::Providers::Azure::CloudManager::Template)
-          timeout_adj = 4
-        elsif target.kind_of?(ManageIQ::Providers::Microsoft::InfraManager::Vm) ||
-              target.kind_of?(ManageIQ::Providers::Microsoft::InfraManager::Template)
-          timeout_adj = 8
-        end
+        timeout_adjustment_multiplier = target.scan_timeout_adjustment_multiplier
```

Comment:
We should probably check that the `target` responds_to `scan_timeout_adjustment_multiplier` since it is only defined for three types of VMs.  (Note this also fixes the spec failure, the target is a `Vmware::InfraManager::Vm` which doesn't define this method)

Reply:
Would the `SupportsFeature` mixin make sense here, or is that overkill?

Reply:
That's probably overkill, if anything I'd probably want to add `:msg_timeout` as an option to the `ost` arg which is set by the scanning mixin.

Reply:
The `scan_timeout_adjustment_multiplier` method is defined in the base class in this PR

Reply:
That would still assume that the target is a VmOrTemplate

Reply:
Thanks - updated!